internal/common/errors.go

```diff
@@ -0,0 +1,50 @@
+package common
+
+import (
+	"errors"
+
+	"github.com/gofiber/fiber/v2"
+	"gorm.io/gorm"
+)
+
+func ServerError(ctx *fiber.Ctx, err error) error {
+	return ctx.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+		"message": err.Error(),
```

Comment:
Using this as an example, but it's valid on the other occurrences as well where we call `.JSON()` for an error.

We should return an [`application/problem+json`](https://datatracker.ietf.org/doc/html/rfc7807) as [per spec](https://developers-italia-4gfxiabty-dip-trasformazione-digitale.vercel.app/it/api/developers-italia), which is something along the lines of:

```json
{
  "errors": [
    {
      "code": "string",
      "title": "string",
      "detail": "string"
    }
  ]
}
```

I think we need a `ProblemJson` struct and I'm also wondering: what if we create a new error (that maybe wraps the `fiber.Error`) and return that from the handlers? The default error handler should take care of it and send the JSON for it.

```go
func PostPublisher(ctx *fiber.Ctx) error {
...
  if (notFound) {
    return commons.NewProblemJson("E001", "Not found", "There's no such publisher")
  }
...
}
```

Also, since `application/problem+json` is a list of errors, will https://github.com/uber-go/multierr be useful?

Reply:
+1 for multierr

Reply:
Sounds cool! Maybe @bfabio you can create an issue for that, what do you think?